lib/govuk_content_models/presentation_toggles.rb

```diff
@@ -0,0 +1,39 @@
+module PresentationToggles
+  extend ActiveSupport::Concern
+
+  included do
+    field :presentation_toggles, type: Hash, default: default_presentation_toggles
+    validates_presence_of :organ_donor_registration_promotion_url, if: :display_organ_donor_registration_promotion?
+  end
+
+  def display_organ_donor_registration_promotion=(value)
+    value = value.is_a?(Boolean) ? value : value != '0' # if assigned using a checkbox
+    organ_donor_registration_promotion_key['display'] = value
+  end
+
+  def display_organ_donor_registration_promotion
+    organ_donor_registration_promotion_key['display']
+  end
+  alias_method :display_organ_donor_registration_promotion?, :display_organ_donor_registration_promotion
+
+  def organ_donor_registration_promotion_url=(value)
+    organ_donor_registration_promotion_key['url'] = value
+  end
+
+  def organ_donor_registration_promotion_url
+    return "" unless display_organ_donor_registration_promotion?
+    organ_donor_registration_promotion_key['url']
+  end
+
+  def organ_donor_registration_promotion_key
+    presentation_toggles['organ_donor_registration_promotion']
+  end
+
+  module ClassMethods
+    def default_presentation_toggles
+      {
+        'organ_donor_registration_promotion' => { 'url' => '', 'display' => false }
+      }
+    end
+  end
+end
```

Comment:
I admire that the way this is stored in the database means it is able to store more toggles without a migration, but it would take a fair bit of code change to support each additional field. Until we actually need it, I think we should do the simplest thing. I suggest simply adding two fields to the `CompletedTransactionEdition` class:
```ruby
  field :organ_donor_registration_promotion_url, type: String
  field :display_organ_donor_registration_promotion, type: Boolean

  validates_presence_of :organ_donor_registration_promotion_url, if: :display_organ_donor_registration_promotion?
```

Two of the tests then fail, but I think they can be removed: `stores organ donor registration promotion URL` fails because the setup tries to create an edition with the display field set, but the URL is empty. In any case, now we've simplified the implementation it seems redundant. `returns an empty organ donor registration URL if the promotion is turned-off`fails, but I think this is really business logic which should be in the presentation layer.